File: 0299-bulls-and-cows/0299-bulls-and-cows.java
class Solution {
    public String getHint(String secret, String guess) {
        int smp[]= new int[10], gmp[]= new int[10];
        int bull=0, cow=0, n= secret.length();
        for(int i=0; i<n; i++){
            char a= secret.charAt(i), b= guess.charAt(i);
            if(a==b) bull++;
            else {
                smp[a-'0']++;
                gmp[b-'0']++;
            }
        }
        for(int i=0; i<10; i++){
            cow+= Math.min(smp[i], gmp[i]);
        }
        return Integer.toString(bull)+ "A"+ Integer.toString(cow)+ "B";
    }
}

File: 0299-bulls-and-cows/0299-bulls-and-cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int bull=0, cow=0;
        unordered_map<char,int> mp;
        for(int i=0; i< secret.size(); i++){
            if(secret[i]==guess[i]){
                guess[i]='g';
                bull++;
            }else mp[secret[i]]++;
        }
        for(int i=0; i<guess.size(); i++){
            if(guess[i]!= 'g'){
                if(mp[guess[i]]>0){
                    mp[guess[i]]--;
                    cow++;
                }
            }
        }
        return to_string(bull)+"A"+ to_string(cow)+ "B";
    }
};